packages/unmock-node/src/backend/client-request-tracker.ts

```diff
@@ -0,0 +1,105 @@
+import debug from "debug";
+import { ClientRequest, IncomingMessage } from "http";
+import _ from "lodash";
+import net from "net";
+import { v4 as uuidv4 } from "uuid";
+
+const debugLog = debug("unmock:client-request-tracker");
+
+const UNMOCK_INTERNAL_HTTP_HEADER = "x-unmock-req-id";
+
+/**
+ * "Static" class for tracking client requests.
+ */
+export default abstract class ClientRequestTracker {
+  public static readonly outgoingRequests: {
+    [requestId: string]: {
+      clientRequest: ClientRequest;
```

Comment:
BTW is this in an object (as opposed to `[requestId: string]: ClientRequest`) in case we'd like to store more items here in the future?

Reply:
Yeah it's a remnant of yesno as they keep track of more objects, could remove one level of nesting there.